src/idea.js

```diff
@@ -6,42 +6,44 @@ class Idea  {
     this.star = star;
     this.comments = [];
   }
-  saveToStorage(idea) {
+  saveToStorage() {
     var retrievedIdeas = JSON.parse(localStorage.getItem("ideas"));
-    if (!retrievedIdeas) {
-      retrievedIdeas = [];
-    } 
-    retrievedIdeas.push(idea);
-    localStorage.setItem("ideas", JSON.stringify(retrievedIdeas));
+    if (retrievedIdeas.length) {
+      for (var i = 0; i < retrievedIdeas.length; i++) {
+        //if retrievedIdeas contains any ideas and the card id already exists, update the card and then update storage
+        if (retrievedIdeas[i].id === this.id) {
```

Comment:
Just wondering if this is necessary as when we are saving a new idea, I'm not sure if there will be an instance in which an idea exists with that exact id already (if the same card already exists, then there might be a problem with our delete function). But if I'm not looking with a wide enough perspective please let me know!

Reply:
This is so that when the idea is starred or commented, it updates (which is why it would have the same id)